agent/site.py

```diff
@@ -459,11 +459,17 @@ def tablewise_backup(self):
         return data
 
     @step("Migrate Site")
-    def migrate(self, skip_failing_patches=False):
+    def migrate(self, skip_search_index=False, skip_failing_patches=False):
+        cmd = "migrate"
+        if skip_search_index:
+            cmd += " --skip-search-index"
         if skip_failing_patches:
-            return self.bench_execute("migrate --skip-failing")
-        else:
-            return self.bench_execute("migrate")
+            cmd += " --skip-failing"
+        return self.bench_execute(cmd)
+
+    @step("Build Search Index")
+    def build_search_index(self):
+        return self.bench_execute("build-search-index")
```

Comment:
This will consume quite a lot of time in update when site is down? 

IMO better way would be to prevent job from running until maintenance mode is disabled.

Reply:
Good point. I'll move it to last then.